### PR TITLE
To do list

### PR DESCRIPTION
**Pull request by MustakimMasum**
Activities of this branch

- Added a crud.js file to add functions to create, read, update, and delete functionality
- Dynamically populated task list section from local storage
- Exported and imported the functions as modules

### CODE REVIEW

File: src/crud.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      taskList.appendChild(listElement);
+    });
+};
+
+deleteTask = (index) => {
+  tasks = tasks.filter((task) => task.index !== index);
+  updateIndex();
+  saveTasks();
+  renderTaskList();
+};
```
Comment (MahmoudBakr23) [inline src/crud.js:89]: - [ ] Delete task function couldn't be tested due to missing UI delete button interaction. Please implement the proper changes in order to fix this issue. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @firepicaso 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on removing the hardcoded tasks and implementing the add new task function
- Descriptive PR and README
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/crud.js (hunk 1 above)
Comment (MustakimMasum) [reply src/crud.js:89]: Hi @MahmoudBakr23 

thank you so much for your feedback. I have the delete button interaction with the 'click' event listener. Lines 48 to 50

`
deleteButton.addEventListener('click', () => {
    deleteTask(task.index);
  });
`

please would you help me understand the issue you mentioned properly.

**Review by danmainah — APPROVED** [review]
Hi @firepicaso , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Implemented the change requested to ensure add, edit and delete functions are working properly.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**